v2/controllers/marketplace/meterbase_controller.go

```diff
@@ -265,6 +297,68 @@ func (r *MeterBaseReconciler) Reconcile(request reconcile.Request) (reconcile.Re
 		instance.Status.Conditions = status.Conditions{}
 	}
 
+	// userWorkloadMonitoringEnabled is considered enabled if both the Spec and cluster configuration are satisfied
+	userWorkloadMonitoringEnabledSpec := instance.Spec.UserWorkloadMonitoringEnabled
+	userWorkloadMonitoringEnabledOnCluster, result := isUserWorkloadMonitoringEnabledOnCluster(cc, r.cfg.Infrastructure, reqLogger)
+	if !result.Is(Continue) {
+		return result.Return()
+	}
+	userWorkloadMonitoringEnabled := userWorkloadMonitoringEnabledOnCluster && userWorkloadMonitoringEnabledSpec
+
+	if instance.Status.Conditions.IsUnknownFor(marketplacev1alpha1.ConditionUserWorkloadMonitoringEnabled) {
+		// Set initial UWM status
+		if result, err := updateUserWorkloadMonitoringEnabledStatus(cc,
+			instance,
+			userWorkloadMonitoringEnabledSpec,
+			userWorkloadMonitoringEnabledOnCluster,
+		); result.Is(Error) || result.Is(Requeue) {
+			if err != nil {
+				return result.ReturnWithError(merrors.Wrap(err, "error updating status"))
+			}
+			return result.Return()
+		}
+	} else if condition := instance.Status.Conditions.GetCondition(marketplacev1alpha1.ConditionUserWorkloadMonitoringEnabled); condition != nil {
```

Comment:
```suggestion
	} 
	
	userWorkloadCondition := instance.Status.Conditions.GetCondition(marketplacev1alpha1.ConditionUserWorkloadMonitoringEnabled)
```

Your first if statement if true should requeue regardless.

Reply:
GetCondition can return nil, not prudent to check since accessing condition.Reason?